homeassistant/components/sensor/__init__.py

```diff
@@ -462,6 +462,26 @@ def capability_attributes(self) -> Mapping[str, Any] | None:
 
         return None
 
+    @property
+    def initial_entity_options(self) -> Mapping[str, Mapping[str, Any]] | None:
+        """Return initial entity options.
+
+        These will be stored in the entity registry the first time the  entity is seen,
+        and then never updated.
+        """
+        initial_unit_of_measurement: str | None = None
+        native_unit_of_measurement = self.native_unit_of_measurement
+
+        if self.device_class == SensorDeviceClass.DISTANCE:
+            initial_unit_of_measurement = self.hass.config.units.get_converted_unit(
+                self.device_class, native_unit_of_measurement
+            )
+
+        if initial_unit_of_measurement is None:
+            return None
+
+        return {DOMAIN: {"initial_unit_of_measurement": initial_unit_of_measurement}}
```

Comment:
This needs to be validated on load that it's still correct, just like when we load sensor_option_unit-of-measurement.